packages/workflow/src/interfaces.ts

```diff
@@ -1,39 +1,164 @@
+   */
+  lastFailure?: TemporalFailure;
 
   /**
-   * Whether a Workflow is replaying history or processing new events
+   * Task queue this Workflow is executing on
    */
-  isReplaying: boolean;
+  taskQueue: string;
+
+  /**
+   * Less commonly used information
+   */
+  more: {
+    /**
+     * Namespace this Workflow is executing in
+     */
+    namespace: string;
+
+    /**
+     * Run Id of the first Run in this Execution Chain
+     */
+    firstExecutionRunId: string;
+
+    /**
+     * The last Run Id in this Execution Chain
+     */
+    continuedFromExecutionRunId?: string;
+
+    // TODO expose from Core
+    /**
+     * Time at which the Workflow Run started
+     */
+    // startTime: Date;
+
+    /**
+     * Milliseconds after which the Workflow Execution is automatically terminated by Temporal Server. Set via {@link WorkflowOptions.workflowExecutionTimeout}.
+     */
+    executionTimeout?: number;
+
+    /**
+     * Time at which the Workflow Execution expires
+     */
+    executionExpirationTime?: Date;
+
+    /**
+     * Milliseconds after which the Workflow Run is automatically terminated by Temporal Server. Set via {@link WorkflowOptions.workflowRunTimeout}.
+     */
+    runTimeout?: number;
+
+    /**
+     * Maximum execution time of a Workflow Task in milliseconds. Set via {@link WorkflowOptions.workflowTaskTimeout}.
+     */
+    taskTimeout: number;
+
+    /**
+     * Retry Policy for this Execution. Set via {@link WorkflowOptions.retry}.
+     */
+    retryPolicy?: RetryPolicy;
+
+    /**
+     * Starts at 1 and increments for every retry if there is a `retryPolicy`
+     */
+    attempt: number;
+
+    /**
+     * Cron Schedule for this Execution. Set via {@link WorkflowOptions.cronSchedule}.
+     */
+    cronSchedule?: string;
+
+    /**
+     * Milliseconds between Cron Runs
+     */
+    cronScheduleToScheduleInterval?: number;
+  };
+
+  /**
+   * These fields are non-deterministic. Workflow code should not do different things based on these fields.
+   */
+  unsafe: {
+    /**
+     * Whether a Workflow is replaying history or processing new events
+     */
+    isReplaying: boolean;
+  };
+}
+
+export interface ParentWorkflowInfo {
+  workflowId: string;
+  runId: string;
+  namespace: string;
+}
+
+// If we get use cases / user requests for this,
+// expose startWorkflow.continueInitiator as WorkflowInfo.initiator: Initiator
+// export enum Initiator {
+//   Unspecified = 0,
+//   ContinueAsNew = 1,
+//   Retry = 2,
+//   Cron = 3,
+// }
+
+export function updateParentType(
+  parent: coresdk.common.INamespacedWorkflowExecution | null | undefined
+): ParentWorkflowInfo | undefined {
+  if (!parent) {
+    return undefined;
+  }
+
+  return {
+    /* eslint-disable @typescript-eslint/no-non-null-assertion */
+    workflowId: parent.workflowId!,
+    runId: parent.runId!,
+    namespace: parent.namespace!,
```

Comment:
Instead of non null assertion let's fail at runtime with an exception or our users will get undefined values if something changes.
In general we shouldn't use non null assertions in the codebase, only in tests.